polkadot/node/network/collator-protocol/src/validator_side_experimental/peer_manager/db.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::validator_side_experimental::{
+	common::Score,
+	peer_manager::{backend::Backend, ReputationUpdate},
+};
+use async_trait::async_trait;
+use polkadot_node_network_protocol::PeerId;
+use polkadot_primitives::{BlockNumber, Hash, Id as ParaId};
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+
+pub struct Db;
+
+// Dummy implementation for now
```

Comment:
I plan on adding a new PR shortly which implements a in-memory DB for testing purposes.
This way, we can test the entire implementation while the DB is still being developed and we can define some unit tests that validate it works as expected

Reply:
Here it is https://github.com/paritytech/polkadot-sdk/pull/8242